File: src/main/java/com/example/demo/model/Owner.java
package com.example.demo.model;

import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Owner {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String firstName;
    private String lastName;
    private String email;

    @OneToMany(mappedBy = "owner")
    private List<Bicycle> bicycles = new ArrayList<>();

    public void addBicycle(Bicycle bicycle) {
        bicycles.add(bicycle);
        bicycle.setOwner(this);
    }

    public void removeBike(Bicycle bicycle) {
        bicycles.remove(bicycle);
        bicycle.setOwner(null);
    }

    // =========================
    // GETTERS & SETTERS
    // =========================

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public List<Bicycle> getBicycles() {
        return bicycles;
    }

    public void setBicycles(List<Bicycle> bicycles) {
        this.bicycles = bicycles;
    }
}

File: src/main/java/com/example/demo/model/Owner.kt
package com.example.demo.model

import jakarta.persistence.*

@Entity
class Owner {
    // =========================
    // GETTERS & SETTERS
    // =========================
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0

    var firstName: String? = null
    var lastName: String? = null
    var email: String? = null

    @OneToMany(mappedBy = "owner")
    var bicycles: MutableList<Bicycle?> = ArrayList<Bicycle?>()

    fun addBicycle(bicycle: Bicycle) {
        bicycles.add(bicycle)
        bicycle.owner= this
    }

    fun removeBike(bicycle: Bicycle) {
        bicycles.remove(bicycle)
        bicycle.owner = null
    }
}